.github/workflows/CI-build.yml

```diff
@@ -1,82 +1,84 @@
-        run: |
-          echo "USER_ID=$(id -u)" >> $GITHUB_ENV
-          echo "GRP_ID=$(id -g)" >> $GITHUB_ENV
+      - name: Upload zkv-node binary artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: "zkv-node"
+          path: target/release/zkv-node
+          retention-days: 1
+          overwrite: true
 
-      - name: Cargo build solo chain
-        run: ${{ env.DOCKER_COMPOSE_CMD }} cargo build -p mainchain --release
+  build-relay-paratest:
+    runs-on: warp-ubuntu-latest-x64-8x
+    name: Build relay chain and paratest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-      - name: Upload solochain runtime artifact
-        uses: actions/upload-artifact@v4
+      - name: Relay chain and paratest build
+        uses: ./.github/actions/cargo-cmd
         with:
-          name: "zkv_runtime.solochain.compact.compressed.wasm"
-          path: target/release/wbuild/zkv-runtime/zkv_runtime.compact.compressed.wasm
-      
-      - name: Cargo build relay chain
-        run: ${{ env.DOCKER_COMPOSE_CMD }} cargo build -p zkv-relay --release
+          command: "cargo build -p zkv-relay -p paratest-node --release --features fast-runtime"
+          use_cache: 'yes'
+          cache_key: 'build-relay-paratest'
+          cmake_install: 'yes'
+          lld_install: 'yes'
 
-      - name: Upload relay chain runtime artifact
+      - name: Upload relay and paratest build binaries
         uses: actions/upload-artifact@v4
         with:
-          name: "zkv_runtime.relaychain.compact.compressed.wasm"
-          path: target/release/wbuild/zkv-runtime/zkv_runtime.compact.compressed.wasm
+          name: "relay-paratest-bins"
+          path: |
+            target/release/zkv-relay
+            target/release/zkv-relay-execute-worker
+            target/release/zkv-relay-prepare-worker
+            target/release/paratest-node
+          retention-days: 1
+          overwrite: true
 
-      - name: Cargo build paratest node
-        run: ${{ env.DOCKER_COMPOSE_CMD }} cargo build -p paratest-node --release
+  build-try-runtime:
+    runs-on: warp-ubuntu-latest-x64-8x
+    name: Build try-runtime
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-      - name: Save build cache
-        if: ${{ !env.ACT && inputs.CACHING_MODE == 'CACHE_SAVE' }}
-        uses: buildjet/cache/save@v3
+      - name: zkv-runtime build
+        uses: ./.github/actions/cargo-cmd
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-            deps/
-          key: build-cache
+          command: "cargo build -p zkv-runtime --release --features try-runtime"
+          use_cache: 'yes'
+          cache_key: 'build-try-runtime'
+          cmake_install: 'yes'
+          lld_install: 'yes'
+
+      - name: Upload try-runtime wasm artifact
```

Comment:
No need for this

Reply:
I guess that here we already have a lot of stuff cached ... Rebuilt it on `CI-try-runtime` means recompile them. Anyway it makes the sequential... but the just try step without compile is really fast (2 minutes) so maybe this approach is a little better

Reply:
Adjusting